class-omnisend-lifterlmsaddon.php

```diff
@@ -0,0 +1,150 @@
+add_action( 'admin_enqueue_scripts', array( 'Omnisend_LifterLMSAddOn', 'load_custom_wp_admin_style' ) );
+
+add_filter( 'plugin_action_links_' . plugin_basename( __FILE__ ), array( 'Omnisend_LifterLMSAddOn', 'add_settings_link' ) );
+
+$omnisend_lifterlms_addon_settings = new SettingsService();
+$omnisend_lifterlms_addon_consent  = new ConsentService();
+
+/**
+ * Class Omnisend_LifterLMSAddOn
+ */
+class Omnisend_LifterLMSAddOn {
+
+	/**
+	 * Register actions for the Omnisend Lifter LMS Add-On.
+	 *
+	 * @param array $actions The array of actions.
+	 *
+	 * @return array The modified array of actions.
+	 */
+	public static function register_actions( array $actions ): array {
+		new OmnisendAddOnAction();
+
+		return $actions;
+	}
+
+	/**
+	 * Autoloader function to load classes dynamically.
+	 *
+	 * @param string $class_name The name of the class to load.
+	 */
+	public static function autoloader( string $class_name ): void {
+		$namespace = 'Omnisend\LifterLMSAddon';
+
+		if ( strpos( $class_name, $namespace ) !== 0 ) {
+			return;
+		}
+
+		$class       = str_replace( $namespace . '\\', '', $class_name );
+		$class_parts = explode( '\\', $class );
+		$class_file  = 'class-' . strtolower( array_pop( $class_parts ) ) . '.php';
+
+		$directory = plugin_dir_path( __FILE__ );
+		$path      = $directory . DIRECTORY_SEPARATOR . 'includes' . DIRECTORY_SEPARATOR . implode( DIRECTORY_SEPARATOR, $class_parts ) . DIRECTORY_SEPARATOR . $class_file;
+
+		if ( file_exists( $path ) ) {
+			require_once $path;
+		}
+	}
+
+	/**
+	 * Check plugin requirements.
+	 */
+	public static function check_plugin_requirements(): void {
+		require_once ABSPATH . '/wp-admin/includes/plugin.php';
+		$lifterlms_addon_plugin = 'omnisend-for-lifterlms-add-on/class-omnisend-lifterlmsaddon.php';
+
+		$omnisend_plugin = 'omnisend/class-omnisend-core-bootstrap.php';
+
+		if ( ! file_exists( WP_PLUGIN_DIR . '/' . $omnisend_plugin ) || ! is_plugin_active( $omnisend_plugin ) ) {
+			deactivate_plugins( $lifterlms_addon_plugin );
+			add_action( 'admin_notices', array( 'Omnisend_LifterLMSAddOn', 'omnisend_notice' ) );
+
+			return;
+		}
+
+		$api_key = get_option( 'omnisend_api_key', null );
+
+		if ( is_null( $api_key ) ) {
+			deactivate_plugins( $lifterlms_addon_plugin );
+			add_action( 'admin_notices', array( 'Omnisend_LifterLMSAddOn', 'omnisend_api_key_notice' ) );
+
+			return;
+		}
+
+		$lifterlms_forms_plugin = 'lifterlms/lifterlms.php';
+
+		if ( ! file_exists( WP_PLUGIN_DIR . '/' . $lifterlms_forms_plugin ) || ! is_plugin_active( $lifterlms_forms_plugin ) ) {
+			deactivate_plugins( $lifterlms_addon_plugin );
+			add_action( 'admin_notices', array( 'Omnisend_LifterLMSAddOn', 'lifterlms_notice' ) );
+		}
+
+		add_action( 'lms_registered_form_actions', array( 'Omnisend_LifterLMSAddOn', 'register_actions' ), 10, 1 );
+	}
+
+	/**
+	 * Display a notice for the missing Omnisend API key.
+	 */
+	public static function omnisend_api_key_notice(): void {
```

Comment:
What user should do "Your Omnisend is not configured properly. Please configure it firstly" ?

Reply:
I change API check if there is any other changes please let me know.